### ISSUE TITLE
Feature request: optimize enemy reinforcements

### ISSUE THREAD

**Issue by Qiou87** [dcs-liberation/dcs_liberation#116]
Hi,

First of all, thanks for the awesome work on this project! I enjoy it a lot.

I noticed that the enemy can have multiple planes sitting in a base and not use them because they are not of the same type - usually because one plane got shot down and the wingman made it back to base. But instead of reinforcing  1x or each during the next turn, the AI will get 2x of the same. So instead of getting 2 patrols by sourcing 2 different planes, they only get one. This is problematic in late game as the AI usually gets run over and would benefit from at least putting up a better fight by optimizing the planes they have. Otherwise after 4-5 turns there is no more CAP (or at best one patrol, too easy to deal with).

**Example**: situation right now
Turn X
1x Su-27
1x Su-30
= 0 patrols

Turn X+1
1x Su-27
1x Su-30
2x Su-33 (new reinforcement)
= 1 patrol of Su-33

**My proposal**:
Turn X+1
2x Su-27 (+1)
2x Su-30 (+1)
= 2 patrols of Su-27 and Su-30

Basically the AI should first try to get even numbers of planes. With the same budget they would be more challenging and interesting to fight.

**Comment by DanAlbert** [comment]
Was all just redone in 2.3. There are a few new bugs filed for follow up work, but it's much smarter now.
